### PR TITLE
chore(android): Change from deprecated user.others to user.data

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description
Refactor setUser native function params naming and underlying native fucntion. 

## :bulb: Motivation and Context
Closes: https://github.com/getsentry/sentry-react-native/issues/2532

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

__#skip-changelog__

### CODE REVIEW

File: android/src/main/java/io/sentry/react/RNSentryModule.java [hunk 1]
```diff
@@ -297,45 +297,45 @@ private static PackageInfo getPackageInfo(Context ctx) {
     }
 
     @ReactMethod
-    public void setUser(final ReadableMap user, final ReadableMap otherUserKeys) {
+    public void setUser(final ReadableMap userKeys, final ReadableMap userDataKeys) {
```
Comment (marandaneto) [inline android/src/main/java/io/sentry/react/RNSentryModule.java:300]: It's missing `segment`.

File: ios/RNSentry.m [hunk 2]
```diff
@@ -296,24 +296,24 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
...
+        if (nil == userKeys && nil == userDataKeys) {
             [scope setUser:nil];
         } else {
             SentryUser* userInstance = [[SentryUser alloc] init];
 
-            if (nil != user) {
-                [userInstance setUserId:user[@"id"]];
-                [userInstance setEmail:user[@"email"]];
-                [userInstance setUsername:user[@"username"]];
+            if (nil != userKeys) {
```
Comment (marandaneto) [inline ios/RNSentry.m:309]: Its missing `IP` and `segment`.

Comment (krystofwoldrich) [reply ios/RNSentry.m:309]: Aaaa, I see, I'll fix that.

File: src/js/wrapper.ts [hunk 3]
```diff
@@ -279,21 +280,23 @@ export const NATIVE: SentryNativeWrapper = {
...
-    let defaultUserKeys = null;
-    let otherUserKeys = null;
+    // separate and serialize all non-default user keys.
+    let userKeys = null;
+    let userDataKeys = null;
     if (user) {
-      const { id, ip_address, email, username, ...otherKeys } = user;
-      defaultUserKeys = this._serializeObject({
-        email,
+      const { id, ip_address, email, username, segment, ...otherKeys } = user;
```
Comment (marandaneto) [inline src/js/wrapper.ts:287]:
```suggestion
      const { id, ip_address, email, username, segment, ...dataKeys } = user;
```

So we get rid of `other`?

**Review by marandaneto — APPROVED** [review]

File: src/js/wrapper.ts (hunk 3 above)
Comment (krystofwoldrich) [reply src/js/wrapper.ts:287]: I just renamed all the variables that at the end are user data, just so it's clear what the variables represents.

**Pull request merged**